source/common/config/watch_map.h

```diff
@@ -109,7 +105,14 @@ class WatchMap : public UntypedConfigUpdateCallbacks, public Logger::Loggable<Lo
                                      Watch* watch);
 
   // Returns the union of watch_interest_[resource_name] and wildcard_watches_.
-  absl::flat_hash_set<Watch*> watchesInterestedIn(const std::string& resource_name);
+  absl::flat_hash_set<Watch*> watchesInterestedIn(const std::string& resource_name,
+                                                  const bool use_prefix_matching);
+
+  std::string prefixFromName(const std::string& resource_name) {
```

Comment:
It seems a bit weird to call this a generic prefix when it's really about `/` prefix. Maybe call it `directory matching` or `namespace matching`.

In any case, I think we will want to rewrite this code a bit for `udpa://` glob collections; we will have a similar logic but need to generalize for both VHDS and globs. @chaoqin-li1123

Reply:
Thanks, I will take that into consideration in the implementation of udpa glob collections.